### DISCUSSION TITLE
texture analysis from DAMASK results

### DISCUSSION THREAD

**Discussion by Erfaaannn** [mtex-toolbox/mtex#1825] [Ask Anything]
Hello

I'm using DAMASK 3.0.0 software for crystal plasticity simulations. After finishing the pre-processing part of my simulation, I got data of orientations as quaternions. I saved these data as a .txt file and now I'm trying to plot pole figures and ODFs from these quaternions. But unfortunately I don't know which of the three buttons 'import pole figure data', 'import EBSD data', 'import ODF data' is the right choice for my purpose. Because MTEX can not work with my text file and in any of the mentioned three buttons, after I import my quaternion text file I face with errors such as:

"Could not detect file format ..."
I thought that I can change the format of my text file to something like .ctf but I don't know how can I do such a thing too. I will be so thankful if someone can help me with my issue

here is my text file for quaternions:
[inc0_HW.txt](https://github.com/mtex-toolbox/mtex/files/11471812/inc0_HW.txt)

**Comment by kilir** [comment]
Hi,
the following should work:
```
quat = dlmread('inc0_HW.txt');
CS = crystalSymmetry('422');
ori = orientation(quat,CS);
% or 
ori = loadOrientation_generic('inc0_HW.txt','CS',CS, 'ColumnNames', ...
      {'Quat real','Quat i','Quat j','Quat k'},'Columns',[1,2,3,4],'Bunge')
 ```
Cheers,
Rdiger

**Reply by Erfaaannn** [reply]
Hi,
Thanks for your attention. I used your codes and now I can import those numerical data into the MATLAB. But now I don't know what should I do next in order to plot pole figures or ODFs. Can you please tell me how can I import a variable like quat in your example into the MTEX GUI?

**Reply by kilir** [reply]
Hi,

in order to plot an odf or a polefigure, have a look at the documentation ( e.g. https://mtex-toolbox.github.io/orientation.calcDensity.html) or or the help e.g. `help odf/plotPDF`.

Loading your file with the gui is not implemented.

Cheers,
Rüdiger
